### PR TITLE
Mkhadiv/bolt humanoid

### PR DESCRIPTION
**Pull request by majidkhadiv**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description
I created urdf for the bolt humanoid on top of bolt urdf. The new robot has one rotational DOF in waist and two DOFs per shoulder in flexion/extension direction

## How I Tested

[//]: # "Explain how you tested your changes"
I created the same demos for the BoltHumanoid as the ones for Bolt. All the demos should be working fine.

### CODE REVIEW

File: demos/demo_display_bolt_humanoid_meshcat.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import pinocchio as pin
+import time
+from robot_properties_bolt.config import BoltHumanoidConfig
+
+if __name__ == "__main__":
+
+    # Load the robot urdf.
+    robot = BoltHumanoidConfig.buildRobotWrapper()
+
+viz = pin.visualize.MeshcatVisualizer(
```
Comment (jviereck) [inline demos/demo_display_bolt_humanoid_meshcat.py]: Should this code and the following be indented under the `if __name__ == "__main__"` statement as well?

File: src/robot_properties_bolt/bolt_humanoid_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+class BoltHumanoidRobot(PinBulletWrapper):
+    def __init__(
+        self,
+        pos=None,
+        orn=None,
+        init_sliders_pose=4 * [0],
+        use_fixed_base=False,
+        is_passive=False
+    ):
+        self.is_passive=is_passive
```
Comment (jviereck) [inline src/robot_properties_bolt/bolt_humanoid_wrapper.py:25]: What does "passive" mean for the robot?

File: src/robot_properties_bolt/resources/dynamic_graph_manager/dgm_parameters_bolt_humanoid.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    # robot status
+    motor_enabled:
+      size: 9
+    motor_ready:
+      size: 9
+    motor_board_enabled:
+      size: 5
+    motor_board_errors:
+      size: 5
+    base_accelerometer:
```
Comment (jviereck) [inline src/robot_properties_bolt/resources/dynamic_graph_manager/dgm_parameters_bolt_humanoid.yaml]: As discussed yesterday, please call the signal `imu_accelerometer`. Same for the other `base_*` signals.

File: src/robot_properties_bolt/bolt_humanoid_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                    leg + "_KFE",
+                ]
+            self.end_eff_ids.append(
+                self.pin_robot.model.getFrameId(leg + "_ANKLE")
+            )
+            self.end_effector_names.append(leg + "_ANKLE")
+
+        # Add waist rotation joint
+        controlled_joints += ["WR"]
+        for arm in ["L", "R"]:
```
Comment (jviereck) [inline src/robot_properties_bolt/bolt_humanoid_wrapper.py:108]: Should we call the arm joints `TL` and `TR` for "top"? We use the prefix `F` as in  `FL` and `FR` as well. Not having strong feelings on this.

**Review by jviereck — CHANGES_REQUESTED** [review]
Small request for changes / discussion. Overall, this looks very good. Thanks for adding this!

**Review by jviereck — CHANGES_REQUESTED** [review]
Overall, this looks quite good. A few small changes than this is good to merge.

Thanks for adding thi!

**Comment by jviereck** [comment]
+\cc @avadesh02 as he might be interested in this as well

File: demos/demo_display_bolt_humanoid_meshcat.py (hunk 1 above)
Comment (MaximilienNaveau) [reply demos/demo_display_bolt_humanoid_meshcat.py]: +1

File: src/robot_properties_bolt/bolt_humanoid_wrapper.py (hunk 2 above)
Comment (MaximilienNaveau) [reply src/robot_properties_bolt/bolt_humanoid_wrapper.py:25]: I think it's passive ankle modeling

Comment (MaximilienNaveau) [reply src/robot_properties_bolt/bolt_humanoid_wrapper.py:25]: opposite to point feet

File: src/robot_properties_bolt/bolt_humanoid_wrapper.py (hunk 4 above)
Comment (MaximilienNaveau) [reply src/robot_properties_bolt/bolt_humanoid_wrapper.py:108]: or "AL" "AR" like 'arm'

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
Overall I agree with @jviereck on the small changes to be made.

File: src/robot_properties_bolt/bolt_humanoid_wrapper.py (hunk 2 above)
Comment (majidkhadiv) [reply src/robot_properties_bolt/bolt_humanoid_wrapper.py:25]: Yes, this is if you wanna also simulate passive ankle later

File: src/robot_properties_bolt/bolt_humanoid_wrapper.py (hunk 4 above)
Comment (majidkhadiv) [reply src/robot_properties_bolt/bolt_humanoid_wrapper.py:108]: but in FL and FR, F stands for front and we do not have sth like front and hind for arms...

**Pull request merged**